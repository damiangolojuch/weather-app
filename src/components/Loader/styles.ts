import styled from "styled-components";
import { Spin } from "antd";

export const LoaderWrapper = styled.div`
  border-radius: 10px;
  background: rgba(255, 255, 255, 0.9);
  text-align: center;
  padding: 200px 0;
`;

export const SpinStyled = styled(Spin)``;
