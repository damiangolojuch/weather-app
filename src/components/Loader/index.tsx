import * as S from "./styles";

const Loader = () => (
  <S.LoaderWrapper>
    <S.SpinStyled />
  </S.LoaderWrapper>
);

export default Loader;
