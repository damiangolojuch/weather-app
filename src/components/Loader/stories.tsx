import { ComponentMeta, ComponentStory } from "@storybook/react";
import Loader from ".";

export default {
  title: "Loader",
  component: Loader,
  args: {
    description: "Simple loader component",
  },
} as ComponentMeta<typeof Loader>;

const DefaultTemplate: ComponentStory<typeof Loader> = () => <Loader />;

export const Default = DefaultTemplate.bind({});
