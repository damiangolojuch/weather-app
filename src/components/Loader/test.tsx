import Loader from ".";
import { render } from "@testing-library/react";

jest.mock("antd", () => {
  const ant = jest.requireActual("antd");
  const Spin = (props: any) => (
    <div data-component="antd/Spin">{props.children}</div>
  );

  return {
    __esModule: true,
    ...ant,
    Spin,
  };
});

const setup = () => {
  return render(<Loader />);
};

describe("<Layout />", () => {
  it("should render antd Spin component with wrapper", () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
});
