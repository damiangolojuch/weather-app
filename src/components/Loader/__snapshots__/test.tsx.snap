// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render antd Spin component with wrapper 1`] = `
<div>
  <div
    class="sc-bczRLJ cpPpMB"
  >
    <div
      data-component="antd/Spin"
    />
  </div>
</div>
`;
